packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -307,7 +308,7 @@ class ScrollPosition extends ViewportOffset {
 
   @override
   void correctBy(double correction) {
-    _pixels += correction;
+    correctPixels(_pixels + correction);
```

Comment:
I would wait until we actually need to do that before making this change. In principle, the change here adds a couple of instructions, so it's not really a simplification. It will also probably end up being different when we do the ScrollPosition split, since some ScrollPositions will not have a `_pixels` value (e.g. leaders that push deltas).

It's also possible that we should just merge the two functions entirely. I don't remember why we ended up with two.